### PR TITLE
Add business logic for login screens

### PR DESCRIPTION
**Pull request by emanuelconunaemme**
Most of the screens require the user to be logged in and we don't want to replicate the logic all over.

Based on the flutter principle "everything is a widget", we are moving this business logic to LoggedScreen. This Widget gets a child widget in the constructor and returns it if the user is logged in otherwise returns the login screen. The login status is accessed via Provider<Login> which is the app root widget.

In order to create a loggedin screen it's sufficient to wrap it with a logged_screen or with a rousseau_logged_scaffold which also gives the toolbar.

### CODE REVIEW

**Comment by panz3r** [comment]
Hi @emanuelconunaemme ,

Looking at your implementation I think that everything is fine 👍 .
A thing I’d change is the name for the “PollDetails” route, I think that using “/polls” can be misleading, something like “poll-details” would be clearer.

Just a little notice about the authentication verification process, most of the auth library provide an async way of checking for the authentication state and this would require having a “Loading” state while the app is checking the token validity (with Keycloak in our case).
A good point in which implement this would be the LoggedScreen you implemented.
Let me know what you think about this.

File: rousseau_vote/lib/src/widgets/rousseau_logged_scaffold.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// toolbar
+class RousseauLoggedScaffold extends StatelessWidget {
+  final Widget body;
+  const RousseauLoggedScaffold(this.body);
+
+  @override
+  Widget build(BuildContext context) {
+    return LoggedScreen(
+        Scaffold(
+          appBar: AppBar(title: Text(AppConstants.TOOLBAR_TITLE)),
```
Comment (giovapanasiti) [inline rousseau_vote/lib/src/widgets/rousseau_logged_scaffold.dart:18]: Shouldn't we give it an optional argument to allow us to override the default title if needed

Comment (emanuelconunaemme) [reply rousseau_vote/lib/src/widgets/rousseau_logged_scaffold.dart:18]: Yes we can do that. I'd add it as far as we need it. We just need to understand how many type of screens we have.

**Review by emanuelconunaemme — COMMENTED** [review]
@panz3r

Yes "poll-details" is better. Also, cause not sure we can distinguish between "/poll" and "/poll/<pollId".

Good idea for the check. Definitely, LoggedScreen seems the right spot.

**Pull request merged**